pandas/_libs/tslibs/timedeltas.pyx

```diff
@@ -1096,25 +1111,30 @@ class Timedelta(_Timedelta):
         # just defer
         if hasattr(other, '_typ'):
             # Series, DataFrame, ...
+            if other._typ == 'dateoffset' and hasattr(other, 'delta'):
+                # Tick offset
+                return self // other.delta
             return NotImplemented
 
-        if hasattr(other, 'dtype'):
+        elif is_timedelta64_object(other):
+            return self // Timedelta(other)
+
+        elif is_integer_object(other) or is_float_object(other):
+            return Timedelta(self.value // other, unit='ns')
+
+        elif hasattr(other, 'dtype'):
+            # ndarray-like; the integer/float object checks exclude
+            # numpy scalars
             if other.dtype.kind == 'm':
                 # also timedelta-like
                 return _broadcast_floordiv_td64(self.value, other, _floordiv)
             elif other.dtype.kind in ['i', 'u', 'f']:
-                if other.ndim == 0:
-                    return Timedelta(self.value // other)
-                else:
-                    return self.to_timedelta64() // other
+                return self.to_timedelta64() // other
```

Comment:
does this not need wrapping in Timedelta?

Reply:
@jorisvandenbossche convinced me that special-casing zero-dim np.ndarrays didn't make much sense.

Reply:
that’s not the point
it’s returning a np.timedelta64 and not a Timedelta

Reply:
OK, will revert.  @jorisvandenbossche if you want to revisit this particular edit, mention it in #17652.